lib/testcase.js

```diff
@@ -185,7 +190,9 @@ TestCase.prototype._run = function () {
   if (this._payload) req.write(this._payload)
   req.end()
 
-  var promise = defer.promise
+  var promise = setupPromise.then(function () {
```

Comment:
did you intend to do the setup and the http request in parallel? i could go either way on this